Binary files vim72.orig/src/.ex_cmds.h.swp and vim72/src/.ex_cmds.h.swp differ
Binary files vim72.orig/src/.ex_docmd.c.swp and vim72/src/.ex_docmd.c.swp differ
diff -r -u -N --exclude='*.orig' vim72.orig/src/ex_cmds.c vim72/src/ex_cmds.c
--- vim72.orig/src/ex_cmds.c	2008-08-04 12:15:00.000000000 -0700
+++ vim72/src/ex_cmds.c	2009-02-18 16:39:35.000000000 -0800
@@ -4021,7 +4021,7 @@
     if (eap->forceit)
 	bigness = curwin->w_height;
     else if (firstwin == lastwin)
-	bigness = curwin->w_p_scr * 2;
+	bigness = p_window_unix2003 ? p_window_unix2003 : curwin->w_p_scr * 2;
     else
 	bigness = curwin->w_height - 3;
     if (bigness < 1)
